src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts

```diff
@@ -383,28 +383,30 @@ export class ExtensionManagementService extends Disposable implements IWorkbench
 
 	private async hasToFlagExtensionsMachineScoped(extensions: IGalleryExtension[]): Promise<boolean> {
 		if (this.isExtensionsSyncEnabled()) {
-			const result = await this.dialogService.show(
-				Severity.Info,
-				extensions.length === 1 ? localize('install extension', "Install Extension") : localize('install extensions', "Install Extensions"),
-				[
-					localize('install', "Install"),
-					localize('install and do no sync', "Install (Do not sync)"),
-					localize('cancel', "Cancel"),
+			const { result } = await this.dialogService.prompt<boolean>({
```

Comment:
@sandy081 fyi please check, this one is a bit more complex adoption of the new dialog api